modules/core/src/smithy4s/Validator.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+sealed trait Validator[A, B] {
+  def validate(value: A): Either[String, B]
+
+  def alsoValidating[C](constraint: C)(implicit
+      ev: RefinementProvider.Simple[C, A]
+  ): Validator[A, B]
+}
+
+object Validator {
+
+  def of[A, B]: ValidatorBuilder[A, B] =
+    new ValidatorBuilder[A, B]
+
+  final class ValidatorBuilder[A, B] private[smithy4s] () {
+    def validating[C](constraint: C)(implicit
+        bijection: Bijection[A, B],
```

Comment:
The bijection should not be provided implicitly here, as it should not be implied

Reply:
Done.